addons/product/models/product.py

```diff
@@ -748,13 +738,15 @@ class SupplierInfo(models.Model):
         help="This vendor's product code will be used when printing a request for quotation. Keep empty to use the internal one.")
     sequence = fields.Integer(
         'Sequence', default=1, help="Assigns the priority to the list of product vendor.")
-    product_uom = fields.Many2one(
-        'uom.uom', 'Unit of Measure',
-        related='product_tmpl_id.uom_po_id',
-        help="This comes from the product form.")
+    uom_purchase_id = fields.Many2one(
```

Comment:
supplier_info / sellers are relative to Purchases then 
```suggestion
    uom_id = fields.Many2one(
```
is sufficient, I suppose

Thanks for this PR solving real use cases

Reply:
I see it less explicit because there are also a new field ``product_uom_id`` that is a related field. So better to mention that this one is for purchase. It can avoid errors during the migration of modules that depends on the old field ``product_uom``.